core/src/main/scala/org/apache/spark/ui/jobs/AllJobsPage.scala

```diff
@@ -17,17 +17,172 @@
 
 package org.apache.spark.ui.jobs
 
-import scala.xml.{Node, NodeSeq}
+import scala.collection.mutable.{HashMap, ListBuffer}
+import scala.xml.{Node, NodeSeq, Unparsed}
 
+import java.util.Date
 import javax.servlet.http.HttpServletRequest
 
-import org.apache.spark.ui.{WebUIPage, UIUtils}
-import org.apache.spark.ui.jobs.UIData.JobUIData
+import org.apache.spark.ui.{UIUtils, WebUIPage}
+import org.apache.spark.ui.jobs.UIData.{ExecutorUIData, JobUIData}
+import org.apache.spark.JobExecutionStatus
 
 /** Page showing list of all ongoing and recently finished jobs */
 private[ui] class AllJobsPage(parent: JobsTab) extends WebUIPage("") {
-  private val startTime: Option[Long] = parent.sc.map(_.startTime)
-  private val listener = parent.listener
+  private val JOBS_LEGEND =
+    <div class="legend-area"><svg width="200px" height="85px">
+      <rect x="5px" y="5px" width="20px" height="15px"
+            rx="2px" ry="2px" stroke="#97B0F8" fill="#D5DDF6"></rect>
+      <text x="35px" y="17px">Succeeded Job</text>
+      <rect x="5px" y="35px" width="20px" height="15px"
+            rx="2px" ry="2px" stroke="#97B0F8" fill="#FF5475"></rect>
+      <text x="35px" y="47px">Failed Job</text>
+      <rect x="5px" y="65px" width="20px" height="15px"
+            rx="2px" ry="2px" stroke="#97B0F8" fill="#FDFFCA"></rect>
+      <text x="35px" y="77px">Running Job</text>
+    </svg></div>.toString.filter(_ != '\n')
+
+  private val EXECUTORS_LEGEND =
+    <div class="legend-area"><svg width="200px" height="55px">
+      <rect x="5px" y="5px" width="20px" height="15px"
+            rx="2px" ry="2px" stroke="#97B0F8" fill="#D5DDF6"></rect>
```

Comment:
Is it possible to create CSS styles that govern this fill? I wanted to play with this coloring but you currently have to go modify the source code everywhere. It would be nice to have styles for each thing here so we could adjust the color in a style sheet.